### ISSUE TITLE
Clear `CurrentAsset` reference without poping up the editor window

### ISSUE THREAD

**Issue by LanKuDot** [JosephBarber96/UnityDialogueEditor#3]
Hi! I found that the editor window kept popping up after recompiling the script. After I researched the code, I created a small patch that checks if there has a opened window before clearing the reference. I thought it could fix the problem. I could create a pull request if needed. Nice asset 😄 !

```diff
@@ -58,6 +58,7 @@ namespace DialogueEditor
         private const string CONTROL_NAME = "DEFAULT_CONTROL";
         public const int MIN_PANEL_WIDTH = 180;
         private const string UNAVAILABLE_DURING_PLAY_TEXT = "Dialogue Editor unavaiable during play mode.";
+        private const string WINDOW_TITLE = "Dialogue Editor";
 
         // Static properties
         public static bool SelectableClickedOnThisUpdate { get; set; }
@@ -104,7 +105,7 @@ namespace DialogueEditor
         [MenuItem("Window/DialogueEditor")]
         public static DialogueEditorWindow ShowWindow()
         {
-            return EditorWindow.GetWindow<DialogueEditorWindow>("Dialogue Editor");
+            return EditorWindow.GetWindow<DialogueEditorWindow>(WINDOW_TITLE);
         }
 
         [UnityEditor.Callbacks.OnOpenAsset(1)]
@@ -392,7 +393,8 @@ namespace DialogueEditor
             // save detection overwriting the object with an empty conversation (save triggerred
             // with no uiNodes present in window due to recompile).
             Log("Scripts reloaded. Clearing current asset.");
-            ShowWindow().CurrentAsset = null;
+            if (HasOpenInstances<DialogueEditorWindow>())
+                GetWindow<DialogueEditorWindow>(WINDOW_TITLE, false).CurrentAsset = null;
         }
```

**Comment by JosephBarber96** [comment]
Hello, thank you for this contribution, it looks great! If you created a pull request, I would happily accept :)

**Comment by LanKuDot** [comment]
Great. But I found that `HasOpenInstances` was added in Unity 2019.3, so it didn't work for the previous version. What is the minimum supported version for this asset?

**Comment by JosephBarber96** [comment]
Currently, DialogueEditor supports 2019 onwards. However, it can likely be safely extended to support previous versions. 

The code in OnScriptsReloaded was originally added to prevent an issue which could cause data loss. However, I put in a fix for this issue here: [Link](https://github.com/JosephBarber96/UnityDialogueEditor/commit/912c6400ab73714c36b7c390030699dfab6fd1b1 )

So now, it seems that `OnScriptsReloaded` can safely be removed altogether, thus the editor window no longer needs to be referenced at all. This has been done on my Localisation feature branch, however has not yet been merged into the Master/Main branch: [Link](https://github.com/JosephBarber96/UnityDialogueEditor/commit/610a34cf2fed33da4e3c68fca2e311bd977ec6dd#diff-d38b85b91a6f5e82d8d560a179aa8cc159c92b29be1c6281dd931faeb8c2071d)

**Comment by LanKuDot** [comment]
Got it. Look forward to the future update.

**Comment by JosephBarber96** [comment]
This has now been fixed in 3e838e2a0642888c059b0e994fceca96d9027568 - thanks!
